### PR TITLE
Redo SessionParamsTuning pass

### PR DESCRIPTION
**Pull request by shaahji**
## Redo SessionParamsTuning pass

Redo the pass to use existing search strategy implementation rather than introduce its own.

Also,
* Renamed the flag "is_generative" to "generative" to match what HF does for model configs.
* ModeBuilder needs to include source model's config in its output. This is required when chaining CLI output with manual run configuration. For CLI -> CLI chain, the model_config.json provides the necessary config.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

**Comment by shaahji** [comment]
NOTE: Tests are still broken for the pass. Posting the changes for the pass itself for early feedback.

File: olive/passes/onnx/session_params_tuning.py [hunk 1]
```diff
@@ -430,149 +404,67 @@
                 "tuning_op_result": tuning_op_result,
             }
 
-    def threads_num_tuning(
+    def get_benchmark(
```
[the remark this answers is not in the document]
Comment (shaahji) [reply olive/passes/onnx/session_params_tuning.py:400]: Resolved!

File: olive/strategy/search_parameter.py [hunk 2]
```diff
@@ -66,6 +69,32 @@
...
+class CategoricalValue(Categorical):
+    """Search parameter that returns user-provided values of parameter as its own support values.
+
+    Examples
+    --------
+    >>> CategoricalValue(default=[1, 2, 3])
+
+    """
+
+    def __init__(self, default: Union[List[str], List[int], List[float], List[bool]]):
```
[the remark this answers is not in the document]
Comment (shaahji) [reply olive/strategy/search_parameter.py]: Resolved!

File: examples/phi3/phi3_tuning.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{
```
Comment (devang-ml) [inline examples/phi3/phi3_tuning.json:1]: Instead of adding a config example, could we just add the `tune-session-params` command line example?

**Review by devang-ml — APPROVED** [review]

**Pull request merged**